robottelo/products.py

```diff
@@ -202,6 +203,14 @@
 from robottelo.cli.repository_set import RepositorySet
 from robottelo.cli.subscription import Subscription
 from robottelo.config import settings
+from robottelo.logging import logger
+
+# The module is now depreciated and will be removed as part of helpers refactor cleanup
+# Whenever an attempt is made to use any classes from this module an excpetion will be raised
+logger.warning(
+    '\033[1;31mThe robottelo.products module is now depreciated!! '
+    'Please use its classes from sat_object.cli_factory object!'
+)
```

Comment:
Since I have replaced all the usage for existing`product.py` in this PR, it is less likely to affect now! Also, this is just a warning, so people can ignore it anyways !

Reply:
There are specific classes that we should use when trying to deprecate things, not just a logged warning in the module.
https://docs.python.org/3/library/exceptions.html#DeprecationWarning

Reply:
Earlier I though to raise an Exception for depreciation but I dont want people to keep completely away from this module lets say in case they wanted to use it for testing purposes or lets say in case of difficulty using the new RepositoryMixins classes.

Reply:
If you've replaced the all current uses of this module in the framework, I think this PR should completely remove the module. Don't leave it here for people to start using again, when there is a 'new' pattern available to use the previous content of the module.

Reply:
That's a good point. Since you've replaced all current uses, then removing the module entirely would be fantastic!